pyproject.toml

```diff
@@ -0,0 +1,122 @@
+# poetry
+[tool.poetry]
+name = "github-profiles-automator"
+version = "0.1.0"
+description = "A charm dedicated to updating Kubeflow Profiles based on a Profiles Management Representation file, that lives in a GitHub repo."
+authors = [
+  "Canonical Analytics Team"
+]
+license = "Apache License 2.0"
+readme = "README.md"
+package-mode = false
+
+[tool.poetry.dependencies]
+python = "^3.12"
+jsonschema = "^4.23.0"
+lightkube = "^0.15.5"
+ops = "^2.17.0"
+
+# Testing tools configuration
+[tool.coverage.run]
+branch = true
+
+[tool.coverage.report]
+show_missing = true
+
+[tool.pytest.ini_options]
+minversion = "6.0"
+log_cli_level = "INFO"
+
+# Linting tools configuration
+[tool.ruff]
+line-length = 99
+lint.select = ["E", "W", "F", "C", "N", "D", "I001"]
+lint.extend-ignore = [
+    "D203",
+    "D204",
+    "D213",
+    "D215",
+    "D400",
+    "D404",
+    "D406",
+    "D407",
+    "D408",
+    "D409",
+    "D413",
+]
+lint.ignore = ["D107"]
+extend-exclude = ["__pycache__", "*.egg_info"]
+lint.per-file-ignores = {"tests/*" = ["D100","D101","D102","D103","D104"]}
+
+[tool.ruff.lint.mccabe]
+max-complexity = 10
+
+[tool.codespell]
+skip = "build,lib,venv,icon.svg,.tox,.git,.mypy_cache,.ruff_cache,.coverage"
+
+# LSPs and formatters
+[tool.pyright]
+include = ["src"]
+
+[tool.black]
+line-length = 99
+target-version = ["py312"]
+
+[tool.isort]
+profile = "black"
+
+# format
+[tool.poetry.group.format]
+optional = true
+
+[tool.poetry.group.format.dependencies]
+ruff = "^0.8.0"
+black = "^24.10.0"
+
+# lint
+[tool.poetry.group.lint]
+optional = true
+
+[tool.poetry.group.lint.dependencies]
+ruff = "^0.8.0"
+codespell = "^2.3.0"
+
+# unit library tests
+[tool.poetry.group.unit-library]
+optional = true
+
+[tool.poetry.group.unit-library.dependencies]
+python = "^3.12"
+coverage = { extras = ["toml"], version = ">7.0" }
```

Comment:
same for jsonschema and lightkube

Reply:
regarding `coverage`, it's not defined in `tool.poetry.dependencies`. Or am I missing something?

Regarding `jsonschema` and `lightkube`, good catch!

Reply:
e76b564

Reply:
Regarding `coverage` I see it in `poetry.lock`, and it seems to be in extras of a lot of packages (under the `test` environment)

So indeed we could omit this, but I think it makes it a bit more clear to see rather than expect to to be downloaded by some package

Reply:
I misunderstood the above, it was not for `coverage` but for `python`. Removed the duplicate python requirement, since it's defined in the main group
cbdc8c4